### PR TITLE
Weed out PROTOTYPE(patterns) comments and patterns.work.md

### PR DESCRIPTION
**Pull request by gafter**
and for each one, either
1. address the issue, or
2. ensure that there is an issue tracking it

This is in preparation for refreshing the pattern-matching code in the `future` branch.

@AlekseyTs @agocke Please review.
/CC @dotnet/roslyn-compiler

### CODE REVIEW

File: docs/features/patterns.md [hunk 1]
```diff
@@ -80,7 +82,7 @@ property_subpattern
     ;
 ```
 
-> Note: There is technically an ambiguity between *type_pattern* (when an *identifier* is absent) and *constant_pattern*, either of which might be a valid parse of a qualified identifier. In practice the ambiguity doesn't matter, and we disambiguate when we bind the name. If we find both a type and a value (e.g. the Color-Color situation), we resolve it to whichever is in the innermost scope. The one exception to this is the `is`-expression, for which we try to bind it as a type for compatibility with previous versions of the language; only if that fails do we resolve it as we do in other contexts.
+> Note: There is technically an ambiguity between *type* in an `is-expression` and *constant_pattern*, either of which might be a valid parse of a qualified identifier. For compatibility we try to bind it as a type for compatibility with previous versions of the language; only if that fails do we resolve it as we do in other contexts, to the first thing found (which must be either a constant or a type).
```
Comment (AlekseyTs) [inline docs/features/patterns.md]: For **compatibility** we try to bind it as a type for **compatibility** with

File: src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs [hunk 2]
```diff
@@ -353,8 +353,8 @@ internal static bool IsQueryPairLambda(SyntaxNode syntax)
         }
 
         /// <summary>
-        /// Returns true if the specified node can represent a closure scope -- that is a scope of a captured variable.
-        /// Doesn't validate whether or not the node actually declares any captured variable.
+        /// Returns true if the specified node is a kind could represent a closure scope -- that is a scope of a captured variable.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs]: It feel s like the new portion is missing some words.

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs [hunk 3]
```diff
@@ -177,7 +177,14 @@ public void M1(object o)
 }
 ";
             CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions)
-            .VerifyDiagnostics()
+            .VerifyDiagnostics(
+                // (13,18): error CS8157: No 'operator is' declaration in 'ArrayList' was found with 1 out parameters
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Perhaps the message should be changed to say "out parameter(s)"?

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs [hunk 4]
```diff
@@ -7684,65 +7704,65 @@ void Test16(int val)
...
+    Diagnostic(ErrorCode.ERR_NameNotInContext, "y10").WithArguments("y10").WithLocation(85, 17),
+    // (87,25): error CS0841: Cannot use local variable 'y10' before it is declared
+    //             case 0 when y10:
+    Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "y10").WithArguments("y10").WithLocation(87, 25),
+    // (89,18): error CS0841: Cannot use local variable 'y10' before it is declared
+    //             case y10:
+    Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "y10").WithArguments("y10").WithLocation(89, 18),
                 // (89,18): error CS0150: A constant value is expected
                 //             case y10:
                 Diagnostic(ErrorCode.ERR_ConstantExpected, "y10").WithLocation(89, 18),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs]: It is not clear why alignment is left inconsistent.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs (hunk 2 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs]: Would it be better to add "of" to make it read as follows?

> Returns true if the specified node is **of** a kind could represent a closure scope -- that is a scope of a captured variable.

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs]: is **of** a kind **that** could

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs (hunk 4 above)
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs]: It is a side-effect of merging. I'll fix it up.It is a side-effect of merging. I'll fix it up.

**Comment by gafter** [comment]
Updated to respond to all comments.

@agocke Please review.

**Comment by agocke** [comment]
:+1:

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**